csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -4,117 +4,274 @@
 using System;
 using System.Text;
 using System.Runtime.InteropServices;
+using System.IO;
 
 namespace Microsoft.ML.OnnxRuntime
 {
     /// <summary>
     /// Holds the options for creating an InferenceSession
     /// </summary>
-    public class SessionOptions:IDisposable
+    public class SessionOptions : IDisposable
     {
-        public IntPtr _nativePtr;
-        protected static readonly Lazy<SessionOptions> _default = new Lazy<SessionOptions>(MakeSessionOptionWithCpuProvider);
+        private IntPtr _nativePtr;
         private static string[] cudaDelayLoadedLibs = { "cublas64_100.dll", "cudnn64_7.dll" };
 
+        #region Constructor and Factory methods
+
         /// <summary>
         /// Constructs an empty SessionOptions
         /// </summary>
         public SessionOptions()
         {
-            NativeMethods.OrtCreateSessionOptions(out _nativePtr);
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateSessionOptions(out _nativePtr));
+        }
+
+
+        private static SessionOptions MakeSessionOptionWithCpuProvider()
```

Comment:
Seems like **MakeSessionOptionWithCpuProvider()** is not referenced anywhere in the project any more.

For Linux OS, the CheckLibcVersionGreaterThanMinimum() function still needs to be run before an InferenceSession is created, to check if the C runtime libraries on the OS are supported by ORT. Previously this check was run in  **MakeSessionOptionWithCpuProvider()** to guarantee the sequence -- does this check still get run somehow?